src/Composer/Plugin/PluginManager.php

```diff
@@ -462,10 +466,28 @@ private function loadRepository(RepositoryInterface $repo, bool $isGlobalRepo):
         }
 
         $sortedPackages = PackageSorter::sortPackages($packages, $weights);
+        if (!$isGlobalRepo) {
+            $requiredPackages = RepositoryUtils::filterRequiredPackages($packages, $rootPackage, true);
+        }
+
         foreach ($sortedPackages as $package) {
             if (!($package instanceof CompletePackage)) {
                 continue;
             }
+
+            if (!in_array($package->getType(), ['composer-plugin', 'composer-installer'],true)) {
+                continue;
+            }
+
+            if (
+                !$isGlobalRepo
+                && !in_array($package, $requiredPackages, true)
+                && !$this->isPluginAllowed($package->getName(), false, true, false)
+            ) {
+                $this->io->writeError('<warning>The "'.$package->getName().'" plugin was not loaded as it is not in the allow-plugins rules and does not seem required by the root package anymore.</warning>');
```

Comment:
```suggestion
                $this->io->writeError('<warning>The "'.$package->getName().'" plugin was not loaded as it is not listed in allow-plugins and is not required by the root package anymore.</warning>');
```

I don't think we call it "allow-plugins rules" anywhere else? And otherwise let's make a definitive statement, either it is required or it's not, no such thing as "seeming required".